File: Back/src/main/java/com/homeExchanger/Home_exchanger/repository/UserRepository.java
package com.homeExchanger.Home_exchanger.repository;

import com.homeExchanger.Home_exchanger.model.Person;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;

public interface UserRepository extends JpaRepository<Person, Long> {
    Optional<Person> findByEmailAddress(String emailAddress);
    List<Person> findBySurname(String surname);
}

File: home-exchange-backend/src/main/kotlin/org/isep/homeexchange/infrastructure/repository/UserRepository.kt
package org.isep.homeexchange.infrastructure.repository

import org.isep.homeexchange.infrastructure.dao.UserDao
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface UserRepository : JpaRepository<UserDao, Long> {
    fun findByEmail(email: String): Optional<UserDao>
}